### PR TITLE
Add additional consent to amp-consent metadata

### PR DESCRIPTION
**Pull request by micajuine-ho**
Allow optional [`additionalConsent`](https://support.google.com/admanager/answer/9681920) to be passed inside optional `metadata` object from CMPs and external iframe flow for amp-consent. 

- Stored in memory within the ConsentMetadataDef (`{"additionalConsent": string|undefined}`)
- Stored in local storage (within metadata object) with `METADATA_STORAGE_KEY` `ac`, if present
    - `"amp-consent:_ping_":{"v":{"s":1,"r":"accept-str","m":{"cst":2,"ac":"1~1.35.41.101"}`
- Available in internal API through `src/consent#getConsentMetadata`
- Sent through the request body of `checkConsentHref` and `onUpdateHref` as well as clientConfig for UI

Tests for: 
- valid, invalid, and undefined 'additionalConsent' being passed in
- mapping between localStorage <==> memory 
- sending correct response body + clientConfig

### CODE REVIEW

File: extensions/amp-consent/0.1/test/test-consent-info.js [hunk 1]
```diff
@@ -150,6 +159,30 @@ describes.fakeWin('ConsentInfo', {}, () => {
...
+      expect(
+        convertStorageMetadata({
+          [METADATA_STORAGE_KEY.CONSENT_STRING_TYPE]:
+            CONSENT_STRING_TYPE.US_PRIVACY_STRING,
+          [METADATA_STORAGE_KEY.ADDITIONAL_CONSENT]: '1~1.35.41.101',
+        })
+      ).to.deep.equal(
+        dict({
+          'consentStringType': CONSENT_STRING_TYPE.US_PRIVACY_STRING,
+          'additionalConsent': '1~1.35.41.101',
```
Comment (zhouyx) [inline extensions/amp-consent/0.1/test/test-consent-info.js:185]: : ) when we add gdprApplies, I think it's sufficient to only test metadata with everything defined. Or there'll be 8 permutations : P

File: extensions/amp-consent/0.1/test/test-consent-ui.js [hunk 2]
```diff
@@ -287,6 +300,10 @@ describes.realWin(
             },
             'consentState': 'accepted',
             'consentStateValue': 'accepted',
+            'consentMetadata': constructMetadata(
```
Comment (zhouyx) [inline extensions/amp-consent/0.1/test/test-consent-ui.js:303]: nit: do you mind rewrite the test, to only check the `promptTrigger` value? Thanks!

File: extensions/amp-consent/0.1/consent-info.js [hunk 3]
```diff
@@ -355,6 +362,10 @@ export function composeMetadataStoreValue(consentInfoMetadata) {
     storageMetadata[METADATA_STORAGE_KEY.CONSENT_STRING_TYPE] =
       consentInfoMetadata['consentStringType'];
   }
+  if (consentInfoMetadata['additionalConsent']) {
```
Comment (zhouyx) [inline extensions/amp-consent/0.1/consent-info.js:365]: reading the spec: looks like the addtl_consent value is never an empty string. Could you please verify? If not, we may need to handle empty string specially.

Comment (micajuine-ho) [reply extensions/amp-consent/0.1/consent-info.js:365]: A valid `addtl_consent` string will never be empty. We do not need to handle this case explicitly.

Comment (zhouyx) [reply extensions/amp-consent/0.1/consent-info.js:365]: Sounds good. to confirm empty string will be converted to `undefined` internally and vendors won't be able to tell the difference

Comment (micajuine-ho) [reply extensions/amp-consent/0.1/consent-info.js:365]: Yup!

**Pull request merged**